tests/lib/sms/src/sms_test.c

```diff
@@ -0,0 +1,1556 @@
+ */
+void test_recv_invalid_udh_portaddr_ignored_concat_valid(void)
+{
+	sms_reg_helper();
+
+	strcpy(test_sms_header.originating_address.address_str, "12345678");
+	test_sms_header.originating_address.length = 8;
+	test_sms_header.originating_address.type = 0x81;
+	test_sms_data.data_len = 15;
+	strcpy(test_sms_data.data, "\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F");
+	test_sms_header.time.year = 21;
+	test_sms_header.time.month = 1;
+	test_sms_header.time.day = 30;
+	test_sms_header.time.hour = 12;
+	test_sms_header.time.minute = 34;
+	test_sms_header.time.second = 56;
+
+	test_sms_header.app_port.present = false;
+
+	test_sms_header.concatenated.present = true;
+	test_sms_header.concatenated.total_msgs = 1;
+	test_sms_header.concatenated.ref_number = 4369;
+	test_sms_header.concatenated.seq_number = 1;
+
+	__wrap_at_cmd_write_ExpectAndReturn("AT+CNMA=1", NULL, 0, NULL, 0);
+	sms_callback_called_expected = true;
+	sms_at_handler(NULL, "+CMT: \"12345678\",22\r\n004408812143658700041210032143652B2C1B01000804111101010400050712345678901234A1061234567890120102030405060708090A0B0C0D0E0F\r\n");
+
+	sms_unreg_helper();
+}
+
+/********* SMS MULTIPLE SEND & RECV TESTS ***********************/
+
+/**
+ * Helper function to provide basic SMS sending.
+ * Useful in tests where multiple send operations are done.
+ */
+void send_basic(void)
+{
+	helper_sms_data_clear();
+
+	enum at_cmd_state state = 0;
+	__wrap_at_cmd_write_ExpectAndReturn("AT+CMGS=15\r0021000A912143658709000003CD771A\x1A", NULL, 0, &state, 0);
+	__wrap_at_cmd_write_IgnoreArg_buf();
+	__wrap_at_cmd_write_IgnoreArg_buf_len();
+
+	int ret = sms_send("+1234567890", "Moi");
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(AT_CMD_OK, state);
+
+	/* Receive SMS-STATUS-REPORT */
+	test_sms_data.type = SMS_TYPE_STATUS_REPORT;
+
+	__wrap_at_cmd_write_ExpectAndReturn("AT+CNMA=1", NULL, 0, NULL, 0);
+	sms_callback_called_expected = true;
+	test_sms_header_exists = false;
+	sms_at_handler(NULL, "+CDS: 24\r\n06550A912143658709122022118314801220221183148000\r\n");
+}
+
+/**
+ * Helper function to provide basic SMS receiving.
+ * Useful in tests where multiple send operations are done.
+ */
+void recv_basic(void)
+{
+	helper_sms_data_clear();
+
+	test_sms_data.type = SMS_TYPE_DELIVER;
+	strcpy(test_sms_header.originating_address.address_str, "1234567890");
+	test_sms_header.originating_address.length = 10;
+	test_sms_header.originating_address.type = 0x91;
+	test_sms_data.data_len = 3;
+	strcpy(test_sms_data.data, "Moi");
+	test_sms_header.time.year = 21;
+	test_sms_header.time.month = 2;
+	test_sms_header.time.day = 9;
+	test_sms_header.time.hour = 20;
+	test_sms_header.time.minute = 58;
+	test_sms_header.time.second = 34;
+
+	__wrap_at_cmd_write_ExpectAndReturn("AT+CNMA=1", NULL, 0, NULL, 0);
+	sms_callback_called_expected = true;
+	sms_at_handler(NULL, "+CMT: \"+1234567890\",22\r\n0791534874894320040A91214365870900001220900285438003CD771A\r\n");
```

Comment:
I think I need help on what to do with this type of lines that have warnings in compliance check:

-:4709: WARNING:LONG_LINE_STRING: line length of 123 exceeds 100 columns
#4709: FILE: tests/lib/sms/src/sms_test.c:1494:
+	sms_at_handler(NULL, "+CMT: \"+1234567890\",22\r\n0791534874894320040A91214365870900001220900285438003CD771A\r\n");

Compliance check is full of this type of warnings for the sms_test.c file. I haven't split any strings into multiple lines to be able to maintain their searchability. So what should be done with these?

Reply:
Linux coding style https://www.kernel.org/doc/html/latest/process/coding-style.html says:
"However, never break user-visible strings such as printk messages because that breaks the ability to grep for them."

Reply:
Discussed these with Robert and I'll split these as nowadays compliance check fails due to these warnings.

Reply:
Actually, compliance check allows long strings without spaces so you need to split the strings from spaces but not in the middle of "words". Otherwise you get another warning. Now all compliance issues are resolved.